sips/sip-45.md

```diff
@@ -0,0 +1,78 @@
+
+| SIP-Number          | 45 |
+| ---:                | :--- |
+| Title               | Prioritized Transaction Submission |
+| Description         | Enable effective prioritized transaction submission. |
+| Author              | Shio Coder \<shio.coder@gmail.com\> |
+| Editor              | Will Riches \<will@sui.io, @wriches\>
+| Type                | Standard |
+| Category            | Core |
+| Created             | 2024-11-26 |
+| Comments-URI        | https://sips.sui.io/comments-45 |
+| Status              | Last Call |
+| Requires            | N/A |
+
+## Abstract
+
+Enable effective prioritized transaction submission based on gas price.
+
+## Motivation
+
+Currently, there is no deterministic gas ordering for the transactions, even when these transactions are submitted within an extremely short time window (less than 100ms). This is due to the fact that different transactions are randomly submitted by different validators, and the validator that is supposed to submit the transaction can have latency jitters. We propose a mechanism to mitigate the random factor and enable prioritized transaction submission.
+
+## Specification
+
+We propose the following changes:
+
+- Increase the maximum allowed value of gas price from $99,999$ to $1,000,000,000,000$ (1T).
+- Allow immediate transaction broadcast based on gas price.
+  - Once a validator receives a transaction with a higher gas price, it is submitted immediately to consensus regardless of tx digest, with the least possible delay, if the validator is in the range of $[0, N)$ of submission position.
+  - $N$ is a function of gas price, defined as:
+
+    - if $`gas\_price < K \times RGP`$ then $N = 1$
+    - else $`N = gas\_price / RGP + 1`$
```

Comment:
What is the downside of just `N = gas\_price / RGP` here? The advantage is that it would be simpler to reason about, and there will be a constant `gas\_price / N` which can be considered as per-submission cost.

Reply:
That works as well!  Originally the thought was to have a slightly higher amplification than simply submitting multiple transactions, so to remove the incentive of doing that.  But the upside is minor and we can definitely choose simplicity over it.

Reply:
That works as well!  Originally the thought was to have a slightly higher amplification than simply submitting multiple transactions, so to remove the incentive of doing that.  But the upside is minor and we can choose simplicity over it.